kubetest2/internal/deployers/eksapi/infra.go

```diff
@@ -84,6 +85,31 @@ func (m *InfrastructureManager) createInfrastructureStack(opts *deployerOptions)
 	if err != nil {
 		return nil, err
 	}
+	// get two AZs for the subnets
+	azs, err := m.clients.EC2().DescribeAvailabilityZones(context.TODO(), &ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return nil, err
+	}
+	var subnetAzs []string
+	if opts.CapacityReservation {
+		subnetAzs, err = m.getAZsWithCapacity(opts)
+		if err != nil {
+			return nil, err
+		}
+		for _, az := range azs.AvailabilityZones {
+			if len(subnetAzs) == 2 {
+				break
+			}
+			if !slices.Contains(subnetAzs, *az.ZoneName) {
+				subnetAzs = append(subnetAzs, *az.ZoneName)
+			}
+		}
+	} else {
+		for i := 0; i < 2; i++ {
+			subnetAzs = append(subnetAzs, *azs.AvailabilityZones[i].ZoneName)
```

Comment:
is the default AZ count guaranteed to be >= 2?

Reply:
I think so, right? Otherwise our original method will also fail.

Reply:
AFAIK, the CFN we have all create with 2 AZs, and new regions I'm aware of also have 3 AZs by default.